### DISCUSSION TITLE
Not able to See the Nft on Metamask

### DISCUSSION THREAD

**Discussion by Sanskar-khandelwal** [Cyfrin/foundry-full-course-cu#2016] [Q&A]
I tried to create my own BasicNft and tried to deploy it. 
Successfully deployed and minted it. But I am not able to see the image on metamask and opensea testnet  (already imported).
But When I deployed the Dog Nft provided by patrick in course I am able to see that 

# Pretending Happy NFT I created and Deployed
Address: ```0xB3668be6a55098485073e226ddc9459E71fad23c```
TokenURI: ```ipfs://bafybeicvfbcqeezq5mqmc3v2cuneqtms2l6fjua5azzsxygebeafl74w7q/?filename=fakesmileuri.json```

# Dogie NFT, That i am able to see on opensea testnet.
Address: ```0xa806AF0705f90B8Fcbd659620D06699FeAc289e0```

**Comment by EngrPips** [comment]
Hello @Sanskar-khandelwal, Please check and confirm that your ipfs `CID` is correct as I cannot view it in a browser using ipfs companion.

**Comment by Sanskar-khandelwal** [comment]
I tried accessing it, and this is the response i get
![image](https://github.com/user-attachments/assets/705aeebf-25db-46e3-9bfe-1a5ccb15d78e)

**Reply by EngrPips** [reply]
Awesome, I was unable to view it. Now try accessing the `ipfs` detail you have in the `image` field of the json.

**Reply by Sanskar-khandelwal** [reply]
yaa even that is working and pointing to the correct image. 

you can read the contract tokenURI function from sepolia etherscan at this address ```0xB3668be6a55098485073e226ddc9459E71fad23c```

**Reply by EngrPips** [reply]
That's weird then that both metamask and opensea isnt fetching it.

**Reply by Sanskar-khandelwal** [reply]
yaa that's weird I tried to figure it out why it is not working, for 2 3 hours.

you can check my own created NFT here: ```0xB3668be6a55098485073e226ddc9459E71fad23c```
and one that  i copied patrick here: ```0xa806AF0705f90B8Fcbd659620D06699FeAc289e0```

Both the contract are verified, maybe try read contract and check tokenUri at 0. 

I tried looking to the format of tokenUri and everything but can't figure out.

**Reply by EngrPips** [reply]
Hello @Sanskar-khandelwal, I think the problem is how you write the `image` field in your `tokenURI.` I noticed Patrick has it differently from yours. write yours like Patrick wrote his, and let's see if there is any change.

**PATRICK TOKENURI**
```js
{
    "name": "PUG",
    "description": "An adorable PUG pup!",
    "image": "https://ipfs.io/ipfs/QmSsYRx3LpDAb1GZQm7zZ1AuHZjfbPkD6J7s9r41xu1mf8?filename=pug.png", <@ here
    "attributes": [
        {
            "trait_type": "cuteness",
            "value": 100
        }
    ]
}
```

**YOUR TOKEN URI**

```js
{
  "attributes": [
    {
      "trait_type": "is Happy",
      "value": "False"
    },
    {
      "trait_type": "Wanna Be Happy",
      "value": "true"
    },
    {
      "trait_type": "Pretending to be happy",
      "value": "True"
    }
  ],
  "description": "A collection of Emoji that has different moods.",
  "image": "ipfs://bafybeietjiuzcwlxw6estokdi2j22k7hawlhydoyonn7l5yabggjrwvbju/fakesmile.png", <@ here
  "name": "Fake Smile Nft"
}
```

**Reply by Sanskar-khandelwal** [reply]
yaa I even tried that, i tried all the way possible by minting this different NFTS
![image](https://github.com/user-attachments/assets/f3dc2d5b-81d0-4a40-b7fe-1f83aeeefb4a)

the Mood NFT that is 100% onchain and decentralized still I am not able to see the image in metamask or opensea
contract address: ```0x03Fd10940B42e63736CAFaBCC7a6e5524caf6a70```
opensea link: ```https://testnets.opensea.io/assets/sepolia/0x03Fd10940B42e63736CAFaBCC7a6e5524caf6a70/0```

I am getting this placeholder image
![image](https://github.com/user-attachments/assets/2a40cac3-1385-4f8c-aa4b-c7a0ecfc6052)

**Reply by EngrPips** [reply]
This is weird. Honestly, I will move on and not get stuck here. I might come back later to figure out what is wrong. I know for sure Metamask had been having issues displaying some `NFTs` properly, but Opensea has been working for many people, so I am not sure what is wrong

**Reply by Sanskar-khandelwal** [reply]
Sure sure, Thanks for the looking into this btw. I saw you are helping a ton of people here. Thank you so much.

**Reply by EngrPips** [reply]
Thanks for the good words; they mean a lot.

**Comment by AntoineF4C5** [comment]
Hi, I wrote a possible solution on another similar discussion, linked [here](https://github.com/Cyfrin/foundry-full-course-cu/discussions/1764#discussioncomment-10056839)

**Reply by EngrPips** [reply]
@Sanskar-khandelwal, please try what @AntoineF4C5 suggested, and let's see if that helps to fix this issue.
